packages/ai-jsx/src/core/conversation.tsx

```diff
@@ -309,3 +323,204 @@ export async function* ShowConversation(
   // we can indicate that we've already yielded the final frame.
   return AI.AppendOnlyStream;
 }
+
+/**
+ * @hidden
+ * "Shrinks" a conversation messages according to a cost function (i.e. token length),
+ * a budget (i.e. context window size), and the `importance` prop set on any `<Shrinkable>`
+ * components within the conversation.
+ *
+ * Currently, `<Shrinkable>` components must wrap conversational messages and have no
+ * effect within the messages themselves.
```

Comment:
Can you be a little more specific about what "no effect" means? Idempotency? Can you add an example?